cluster/gce/util.sh

```diff
@@ -325,6 +325,19 @@ function detect-master () {
   echo "Using master: $KUBE_MASTER (external IP: $KUBE_MASTER_IP)"
 }
 
+# Reads kube-env metadata from master
+#
+# Assumed vars:
+#   KUBE_MASTER
+#   PROJECT
+#   ZONE
+function get-master-env() {
+  # TODO(zmerlynn): Make this more reliable with retries.
```

Comment:
This is "liberated" from the upgrade script yes.  I hope that if we get this merged, then maybe we can even get the upgrade script working again in kube-up.  Maybe :-)  cc @zmerlynn